std/variant.d

```diff
@@ -775,58 +776,19 @@ public:
      * VariantException).
      */
 
-    @property T get(T)() if (!is(T == const))
+    @property inout(T) get(T)() inout
     {
-        auto p = *cast(T**) &store;
-
-        /* handler(OpID.get, ) expects the TypeInfo for T in the same buffer it
-         * writes the result to afterwards. Because T might have a non-trivial
-         * destructor, postblit or invariant, we cannot use a union.
-         */
-        struct Buf
-        {
-            T result;
-            // Make sure Buf.sizeof is big enough to store a TypeInfo in
-            void[T.sizeof < TypeInfo.sizeof ? TypeInfo.sizeof - T.sizeof : 0] init = void;
-        }
-        TypeInfo info = typeid(T);
-        Buf buf;
-        memcpy(&buf, &info, info.sizeof);
-
-        if (fptr(OpID.get, &store, &buf))
-        {
-            throw new VariantException(type, typeid(T));
-        }
-        return buf.result;
-    }
-
-    @property T get(T)() const if (is(T == const))
-    {
-        auto p = *cast(T**) &store;
-
-        /* handler(OpID.get, ) expects the TypeInfo for T in the same buffer it
-         * writes the result to afterwards. Because T might have a non-trivial
-         * destructor, postblit or invariant, we cannot use a union.
-         */
-        struct Buf
-        {
-            static if (is(T == shared))
-                shared(Unqual!T) result;
-            else
-                Unqual!T result;
-
-            // Make sure Buf.sizeof is big enough to store a TypeInfo in
-            void[T.sizeof < TypeInfo.sizeof ? TypeInfo.sizeof - T.sizeof : 0] init = void;
-        }
-        TypeInfo info = typeid(T);
-        Buf buf;
-        memcpy(&buf, &info, info.sizeof);
+        static if (is(T == shared))
+            shared Unqual!T result;
+        else
+            Unqual!T result;
```

Comment:
> I think this can be = void, is that right?

I don't see how it could be. `result` should be properly default initialized now. Where would an uninitialized value come from?